freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/user/CloudIdentitySyncService.java

```diff
@@ -48,16 +58,49 @@ private void syncAzureObjectIds(Stack stack, UmsUsersState umsUsersState, BiCons
 
             SetRangerCloudIdentityMappingRequest setRangerCloudIdentityMappingRequest = new SetRangerCloudIdentityMappingRequest();
             setRangerCloudIdentityMappingRequest.setAzureUserMapping(userToAzureObjectIdMap);
-            // TODO The SDX endpoint currently sets the config and triggers refresh. The SDX endpoint should also be updated
-            //      to allow polling the status of the refresh.
+
             LOGGER.debug("Setting ranger cloud identity mapping: {}", setRangerCloudIdentityMappingRequest);
-            sdxEndpoint.setRangerCloudIdentityMapping(envCrn, setRangerCloudIdentityMappingRequest);
+            RangerCloudIdentitySyncStatus syncStatus = sdxEndpoint.setRangerCloudIdentityMapping(envCrn, setRangerCloudIdentityMappingRequest);
+
+            // The sync status represents a cloud identity sync that may still be in progress, which we need to poll to check for completion.
+            pollSyncStatus(syncStatus, envCrn, warnings);
         } catch (Exception e) {
             LOGGER.warn("Failed to set cloud identity mapping for environment {}", envCrn, e);
             warnings.accept(envCrn, "Failed to set cloud identity mapping");
         }
     }
 
+    private void pollSyncStatus(RangerCloudIdentitySyncStatus syncStatus, String envCrn, BiConsumer<String, String> warnings) {
+        // NOTE: Although it's synchronously polling, in practice this sync takes less than a second to complete
+        Instant startTime = clock.getCurrentInstant();
+        while (true) {
+            LOGGER.info("syncStatus = {}", syncStatus);
+            switch (syncStatus.getState()) {
+                case SUCCESS:
+                    LOGGER.info("Successfully synced cloud identity, envCrn = {}", envCrn);
+                    return;
+                case FAILED:
+                    LOGGER.error("Failed to sync cloud identity, envCrn = {}", envCrn);
+                    warnings.accept(envCrn, "Failed to sync cloud identity into environment");
+                    return;
+                default:
+                    LOGGER.info("Sync is in progress, retrying");
+                    try {
+                        Thread.sleep(SYNC_SLEEP_INTERVAL_MS);
+                    } catch (InterruptedException e) {
+                        LOGGER.error("Interrupted during cloud identity sync", e);
+                        warnings.accept(envCrn, "Interrupted during cloud identity sync");
+                    }
+                    long commandId = syncStatus.getCommandId();
+                    syncStatus = sdxEndpoint.getRangerCloudIdentitySyncStatus(envCrn, commandId);
```

Comment:
it's possible to make this call and then not check the returned value because we timed out. I suggest moving the clock isAfter check to before the sleep.

Reply:
hmm true, I've moved the check up higher.